### PR TITLE
feat: Adding the Terraform module

### PR DESCRIPTION
**Pull request by gatici**
# Description

This PR adds the Terraform module for the charm. Hence, the charm could be deployed using Terraform.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

File: terraform/.gitignore
Comment (markbeierl) [inline terraform/.gitignore:1]: Shouldn't this file be in a top level directory?

Comment (gatici) [reply terraform/.gitignore:1]: That .gitignore file includes the things for Terraform. It's possible to merge the contents in the top level directory or we can keep them separate. Should we move them ?

Comment (gatici) [reply terraform/.gitignore:1]: Done

File: terraform/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+- **output.tf** - Responsible for integrating the module with other Terraform modules, primarily by defining potential integration endpoints (charm integrations), but also by exposing the application name.
+- **terraform.tf** - Defines the Terraform provider.
+
+## Using sdcore-udr-k8s base module in higher level modules
+
+If you want to use `sdcore-udr-k8s` base module as part of your Terraform module, import it like shown below.
+
+```text
+module "sdcore-udr-k8s" {
+  source                 = "git::https://github.com/canonical/sdcore-udr-k8s-operator//terraform"
```
Comment (Gmerold) [inline terraform/README.md:22]: Why double `/`? Single `/` works too.

**Review by Gmerold — APPROVED** [review]

**Pull request merged**